middleware/http/oauth2/oauth2_middleware.go

```diff
@@ -83,7 +83,7 @@ func (m *Middleware) GetHandler(ctx context.Context, metadata middleware.Metadat
 			session := sessions.Start(w, r)
 
 			if session.GetString(meta.AuthHeaderName) != "" {
-				w.Header().Add(meta.AuthHeaderName, session.GetString(meta.AuthHeaderName))
+				r.Header.Add(meta.AuthHeaderName, session.GetString(meta.AuthHeaderName))
```

Comment:
We have another place as well, where this change would be required: https://github.com/dapr/components-contrib/blob/907b6b9ebeeba9c4f74f4df1ae3316eae3cb784e/middleware/http/oauth2/oauth2_middleware.go#L139

Can we have a unit test for it, so that we don't get this issue at least again?

Reply:
> We have another place as well, where this change would be required:
> 
> https://github.com/dapr/components-contrib/blob/907b6b9ebeeba9c4f74f4df1ae3316eae3cb784e/middleware/http/oauth2/oauth2_middleware.go#L139
> 
> Can we have a unit test for it, so that we don't get this issue at least again?

I was thinking the same but it was going to require injecting a mock for getting tokens (similar to oauth2clientcredentials component). Its a more significant code change. I'll give it a go today and update the PR.

Reply:
FYI. 
I updated the second location where header is added. Thanks for pointing that out. 
I created a test for the first case which doesn't require a Mock. 
I'll work on the second case which requires a Mock.

Reply:
@DeepanshuA There are tests added for both places where a Request header is added.

Do you see a purpose for the Header being added after the token Exchange? The next line sends a redirect response to the client, so the request header has no purpose. 

The header (before the redirect) has been added that way all the back to v0.1, but I'm not sure the reason.

Reply:
I agree the header on (current) L148 is irrelevant

Reply:
I removed the unneeded header and the test.

Reply:
Yup, agreed - it is a redirect - I missed that. request header is not required at that place then.